Automattic/wp-e2e-tests
# Add tests for 2fa login

This PR adds 4 simple tests for 2fa login:
- email/password login with 2fa code via application (otp)
- email/password login with 2fa code via sms
- passwordless login with 2fa code via application (otp)
- passwordless login with 2fa code via sms

This required the addition of 2 new features:
- `lib/xmpp-client.js` which is used to listen to new sms for an account via xmpp.
- [`speakeasy`](https://github.com/speakeasyjs/speakeasy) dependency, which is a lib for OTP.

The code is a bit messy and could be refactored. I suggest we do this is a follow up PR.

### Testing Instructions
- `git checkout update/2fa-login`
- Create/Modify `config/local-test.json`  to add the new accounts information (see pb/1ce96). You can also use `local-development.json` if you have calypso running locally and `public-api.wordpress.com` sandboxed with the right filter to rewrite magic link urls).
- Run it for 1 device: `NODE_ENV=test BROWSERSIZE=mobile ./node_modules/.bin/mocha specs/wp-log-in-out-spec.js`
- Try multiple devices at once: `NODE_ENV=test ./node_modules/.bin/mocha specs/wp-log-in-out-spec.js`

### Reviews
